File: backend/internal/dto/subject.go
package dto

type SubjectResp struct {
	ID          uint   `json:"id"`
	LevelID     uint   `json:"level_id"`
	Name        string `json:"name"`
	ShortName   string `json:"short_name"`
	Description string `json:"description"`
	Icon        string `json:"icon"`
	SortOrder   int    `json:"sort_order"`
}

type SubSubjectResp struct {
	ID        uint   `json:"id"`
	SubjectID uint   `json:"subject_id"`
	Name      string `json:"name"`
	SortOrder int    `json:"sort_order"`
}

type ChapterResp struct {
	ID            uint   `json:"id"`
	SubjectID     uint   `json:"subject_id"`
	SubSubjectID  uint   `json:"sub_subject_id"`
	ParentID      uint   `json:"parent_id"`
	Name          string `json:"name"`
	SortOrder     int    `json:"sort_order"`
	QuestionCount int64  `json:"question_count"`
	MaterialID    uint   `json:"material_id"`
}

File: frontend/src/types/subject.ts
export interface ExamLevel {
  id: number
  name: string
}

export interface Subject {
  id: number
  level_id: number
  name: string
  short_name: string
  description: string
  icon: string
  sort_order: number
}

export interface SubSubject {
  id: number
  subject_id: number
  name: string
  sort_order: number
}

export interface Chapter {
  id: number
  subject_id: number
  sub_subject_id: number
  parent_id: number
  name: string
  sort_order: number
  question_count: number
  material_id: number
}
